docs/guide/using-your-ide.md

```diff
@@ -0,0 +1,243 @@
+## Using Your IDE
+
+Using an [Integrated Development Environments (IDE)](https://en.wikipedia.org/wiki/Integrated_development_environment) is for sure convenient for its features (graphical debugger, edition tools, etc.).
+You may be interested in using your favorite IDE to develop a Webots controller.
+
+A priori, Webots can support any IDE to create, build and debug its controllers.
+It is simply a matter of setting up the IDE correctly to fulfill the Webots controller build rules.
+
+Documenting every IDE for each OS is a huge and unmaintainable task.
+This is why only some of them are documented.
+They could serve as a reference for others.
+
+The build rules are:
+
+- The target executable file should have the same name as the controller directory, and follow strictly this path: `$WEBOTS_PROJECT/controllers/$(CONTROLLER_NAME)/$(CONTROLLER_NAME)$(EXE_EXTENSION)`.
+- The executable should be linked with the Webots controller library (and eventually include the corresponding path):
```

Comment:
Unfortunately, this is not sufficient with Visual C++ where we need to recompile all the libCppController files with the controller...